### DISCUSSION TITLE
Inheritance of type var bounds?

### DISCUSSION THREAD

**Discussion by BaconPancakes** [microsoft/pyright#8399] [Q&A]
If a type variable of a parent generic is bound to a type, shouldn't derived classes of that type be subject to the same bounds?

Consider the following behavior:

Code sample in [pyright playground](https://pyright-play.net/?pythonVersion=3.12&strict=true&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoASAhgM4AWRARgDYCmAsAFBNMDG1pJUAQqbQOK0UtEElYBtACoAuKAApi5KnQCUAXWlMo2qADp9LRu048wAVxQATXiQFCRYuTbvDRE6khIw1K2QGJYRFooEXAQKFIQgA8EWlYYWktZDy9MLhQwfAolGgZGHT0DZiMOEi4AEQcAN0TBVzEpNSc%2BOocJSR9-QNiQkDCoDPwEEFpbFHxaGpRYMnM0Mh4W%2BzdGtKhKcytYMChsilytHX1dQ2My0wtLStEay1aVlO8u4bBciCIYMXW4ojNbRds9zEAHIuPAelUiOFhIkwTtKME9so8gVjkwgA)

```python
from typing import Hashable


class BaseGeneric[T: (Hashable)]:
    ...

class BoundBaseGeneric(BaseGeneric[list]): # type error as expected: list is not hashable
    ...

class DerivedGeneric[T](BaseGeneric[T]): # type error not present even though BaseGeneric[T] is bound to hashable
    ...

class BoundDerivedGeneric[list]: # problematic because BaseGeneric's type var needs to be hashable
    ...

```

**Comment by erictraut** [comment]
Yes, the bound should be enforced. Thanks for reporting. Converting the discussion to a bug report.

**Reply by erictraut** [reply]
Here's a link to the issue: https://github.com/microsoft/pyright/issues/8400
